packages/fxa-settings/src/components/Settings/LinkedAccounts/index.test.tsx

```diff
@@ -26,14 +42,33 @@ const clickConfirmUnlinkButton = async () => {
     fireEvent.click(confirmButton);
   });
 };
+
+// This directly modifies `account.linkedAccounts` for the account object
```

Comment:
This was kind of gnarly to figure out for a few reasons.

Before these changes, we were testing for `expect(screen.queryAllByTestId('linked-account')).toHaveLength(0);` in our `on unlink, removes linked account` test. This passed because we didn't have any elements in the codebase named this - it should have been `settings-linked-account`, which would have failed.

It looks like we understandably used `ConnectedServices` tests as a base here for the original setup, but I think a combination of what we're testing over there and the component rerendering by default on change (I _think_ because of our `sortedAndUniqueClients` function) made that same setup not work here. This test setup basically simulates what we're doing in the apollo client cache when `account.mockUnlinkThirdParty` is called, except we're directly mutating the object passed into it due to the way that we mock `account`.